tests/unit/Location.cpp

```diff
@@ -0,0 +1,128 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#define BOOST_TEST_MODULE location
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/Model/Binary.h"
+#include "revng/Pipes/Locations.h"
+
+namespace location = revng::pipes::location;
+
+BOOST_AUTO_TEST_SUITE(RevngLocationInfrastructure);
+
+BOOST_AUTO_TEST_CASE(MetaAddressAsTheKey) {
+  using namespace location;
+  auto Instr = Instruction.make(MetaAddress::fromString("0x123:Generic64"),
+                                MetaAddress::fromString("0x456:Generic64"),
+                                MetaAddress::fromString("0x789:Generic64"));
+
+  revng_check(std::get<"basic-block">(Instr).address() == 0x456);
+
+  std::get<"basic-block">(Instr) += std::get<"address">(Instr).address();
+
+  revng_check(std::get<"basic-block">(Instr).address() == 0xbdf);
+
+  constexpr auto Expected = "/instruction/0x123:Generic64/"
+                            "0xbdf:Generic64/0x789:Generic64";
+  revng_check(Instr.toString() == Expected);
+  revng_check(Instr.toString() == serializeToString(Instr));
+}
+
+static model::TypePath makeFunction(model::Binary &Model) {
+  model::CABIFunctionType Function;
+  Function.CustomName = "my_cool_func";
+  Function.OriginalName = "Function_at_0x40012f:Code_x86_64";
+  Function.ABI = model::ABI::SystemV_x86_64;
+
+  using UT = model::UpcastableType;
+  auto Ptr = UT::make<model::CABIFunctionType>(std::move(Function));
+
+  return Model.recordNewType(std::move(Ptr));
+}
+
+BOOST_AUTO_TEST_CASE(TypeIDAsTheKey) {
+  TupleTree<model::Binary> NewModel;
+  auto FunctionPath = makeFunction(*NewModel);
+
+  auto Field = revng::pipes::location::TypeField.make();
+  std::get<"type-key">(Field) = FunctionPath.get()->key();
+  std::get<"field-index">(Field) = 2; // And arbitrary field index. WIP
```

Comment:
WIP?

Reply:
Something akin to "I'm not sure this should be an integer-index but I can't think of anything better right now, should look into this later".

Reply:
This is also gone.